server/aesdsocket.c

```diff
@@ -268,146 +268,175 @@ void * log_timestamp_write()
-            /*in case of write failure close the fd connection for the client and free the memory */
-            if ((ret_status = write(fd_wr_file, client_node->malloc_wr_buffer, malloc_buffer_len)) < 0)
+
+            if (!memcmp(client_node->malloc_wr_buffer,ioc_seek, 19))
             {
-                free(client_node->malloc_wr_buffer);
-                client_node->malloc_wr_buffer = NULL;
-                close(client_node->fd);
-                syslog(LOG_INFO, "Closed connection from %s", ip4addr_str);
-                pthread_mutex_unlock(&lock);
-                return NULL;
+                wr_cmd = memchr(client_node->malloc_wr_buffer, ':', malloc_buffer_len);
+
+                if ((client_node->malloc_wr_buffer - wr_cmd) >= (malloc_buffer_len - 1))
+                {
+                    free(client_node->malloc_wr_buffer);
+                    client_node->malloc_wr_buffer = NULL;
+                    close(file_fd);
+                    close(client_node->fd);
+                    syslog(LOG_INFO, "Closed connection from client");
+                    pthread_mutex_unlock(&lock);
+                    return NULL;
+                }
+
+                wr_offset = memchr(client_node->malloc_wr_buffer, ',', malloc_buffer_len);
+
+                if ((client_node->malloc_wr_buffer - wr_offset) >= (malloc_buffer_len - 1))
+                {
+                    free(client_node->malloc_wr_buffer);
+                    client_node->malloc_wr_buffer = NULL;
+                    close(file_fd);
+                    close(client_node->fd);
+                    syslog(LOG_INFO, "Closed connection from client");
+                    pthread_mutex_unlock(&lock);
+                    return NULL;
+                }
+
+                wr_cmd++;
+
+                if (wr_cmd == wr_offset)
+                {
+                    free(client_node->malloc_wr_buffer);
+                    client_node->malloc_wr_buffer = NULL;
+                    close(file_fd);
+                    close(client_node->fd);
+                    syslog(LOG_INFO, "Closed connection from client");
+                    pthread_mutex_unlock(&lock);
+                    return NULL;
+                }
+
+                *wr_offset = '\0';
+
+                seekto.write_cmd = atoi(wr_cmd);
+
+                wr_offset++;
+
+                if (wr_offset == client_node->malloc_wr_buffer + (malloc_buffer_len - 1))
+                {
+                    free(client_node->malloc_wr_buffer);
+                    client_node->malloc_wr_buffer = NULL;
+                    close(file_fd);
+                    close(client_node->fd);
+                    syslog(LOG_INFO, "Closed connection from client");
+                    pthread_mutex_unlock(&lock);
+                    return NULL;
+                }
+
+                client_node->malloc_wr_buffer[malloc_buffer_len - 1] = '\0';
+
+                seekto.write_cmd_offset = atoi(wr_offset);
```

Comment:
@Nileshkartik95 what does this assume about the number of characters are `wr_offset`?